survey.md

```diff
@@ -1,55 +1,51 @@
-## Brief
+## Overview
 
-This document will introduce some features or design of customizing fuzzer. Firstly, most of fuzzer implemented its own Genetic Algorithm( GA). Some features can be classified to one of GA component. For example, the optimizing of generate, mutate and crossover. Other features, such as special feedback or satifying deep nested condition, is strongly depend on what project you fuzz, although these problem is very common in real-world project.
+This document will introduce some features or design of the customized fuzzer. Firstly, most of fuzzer implemented its own Genetic Algorithm (GA). Some features can be classified into one of the GA components. For example, the optimizing of generating, mutate and crossover. Other features, such as special feedback or satisfying deep nested condition, strongly depend on what project you fuzz, although this problem is very common in real-world projects.
 
-Because this document is a by-product of customizing Linux kernel fuzzer(base on Syzkaller), Some problem appeared kernel fuzzing only. At the end this document, i will attach the paper the document involved, with a short introduction.  
+Because this document is a by-product of customizing Linux kernel fuzzer(base on Syzkaller), Some problems appeared in kernel fuzzing only. At the end of this document, I will attach the paper to the document involved, with a short introduction.  
 
 
 ## GA of fuzzer
 
-In most fuzzers, GA is the engine of evolving testcase. For different purpose, the design of GA's components can be quite different.
-
+In most fuzzers, GA is the engine of evolving testcase. For different purposes, the design of GA's components can be quite different.
 
 ### Generate & Mutate in evaluating programming
 
-In evolutionary programming, if mutation and generating only base on random inputs, that fuzzer will perform badly. Useful information help reducing the search space of evolving the testcase you want. Generally, these following informations can benefit mutating or generating:
-1. symbolic execution: static analyse target, deriver which inputs is useful.( KLEE)
-2. Dynamically taint analysis( DTA): Dynamically trace and derive which input satisfy which conditions efficiently.( Vuzzer)
-3. Manually write manner: hard-code some special inputs or enum inputs.( Syzkaller)
-4. Extract input from real-world program( Moonshine).
+In evolutionary programming, if mutation and generating only base on random inputs, that fuzzer will perform badly. Useful information help to reduce the search space of evolving the testcase you want. Generally, these following information can benefit mutating or generating:
+1. symbolic execution: static analysis target, deriver which inputs are useful. (KLEE)
```

Comment:
Keep all list like:
1.  symbolic execution: static analysis target, deriver which inputs are useful.( KLEE)
"deriver" should be "derive"